File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/musicbrainz/artist/get/ArtistGet.java
package uk.co.ourfriendirony.medianotifier.clients.musicbrainz.artist.get;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "disambiguation",
        "id",
        "area",
        "gender",
        "begin-area",
        "sort-name",
        "gender-id",
        "name",
        "type",
        "release-groups",
        "end-area",
        "ipis",
        "type-id",
        "life-span",
        "isnis",
        "country"
})
public class ArtistGet {
    @JsonProperty("disambiguation")
    private String disambiguation;
    @JsonProperty("id")
    private String id;
    @JsonProperty("area")
    private ArtistGetArea area;
    @JsonProperty("gender")
    private Object gender;
    @JsonProperty("begin-area")
    private ArtistGetBeginArea beginArea;
    @JsonProperty("sort-name")
    private String sortName;
    @JsonProperty("gender-id")
    private Object genderId;
    @JsonProperty("name")
    private String name;
    @JsonProperty("type")
    private String type;
    @JsonProperty("release-groups")
    private List<ArtistGetReleaseGroup> releaseGroups = null;
    @JsonProperty("end-area")
    private Object endArea;
    @JsonProperty("ipis")
    private List<Object> ipis = null;
    @JsonProperty("type-id")
    private String typeId;
    @JsonProperty("life-span")
    private ArtistGetLifeSpan lifeSpan;
    @JsonProperty("isnis")
    private List<String> isnis = null;
    @JsonProperty("country")
    private String country;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("disambiguation")
    public String getDisambiguation() {
        return disambiguation;
    }

    @JsonProperty("disambiguation")
    public void setDisambiguation(String disambiguation) {
        this.disambiguation = disambiguation;
    }

    @JsonProperty("id")
    public String getId() {
        return id;
    }

    @JsonProperty("id")
    public void setId(String id) {
        this.id = id;
    }

    @JsonProperty("area")
    public ArtistGetArea getArea() {
        return area;
    }

    @JsonProperty("area")
    public void setArea(ArtistGetArea area) {
        this.area = area;
    }

    @JsonProperty("gender")
    public Object getGender() {
        return gender;
    }

    @JsonProperty("gender")
    public void setGender(Object gender) {
        this.gender = gender;
    }

    @JsonProperty("begin-area")
    public ArtistGetBeginArea getBeginArea() {
        return beginArea;
    }

    @JsonProperty("begin-area")
    public void setBeginArea(ArtistGetBeginArea beginArea) {
        this.beginArea = beginArea;
    }

    @JsonProperty("sort-name")
    public String getSortName() {
        return sortName;
    }

    @JsonProperty("sort-name")
    public void setSortName(String sortName) {
        this.sortName = sortName;
    }

    @JsonProperty("gender-id")
    public Object getGenderId() {
        return genderId;
    }

    @JsonProperty("gender-id")
    public void setGenderId(Object genderId) {
        this.genderId = genderId;
    }

    @JsonProperty("name")
    public String getName() {
        return name;
    }

    @JsonProperty("name")
    public void setName(String name) {
        this.name = name;
    }

    @JsonProperty("type")
    public String getType() {
        return type;
    }

    @JsonProperty("type")
    public void setType(String type) {
        this.type = type;
    }

    @JsonProperty("release-groups")
    public List<ArtistGetReleaseGroup> getReleaseGroups() {
        return releaseGroups;
    }

    @JsonProperty("release-groups")
    public void setReleaseGroups(List<ArtistGetReleaseGroup> releaseGroups) {
        this.releaseGroups = releaseGroups;
    }

    @JsonProperty("end-area")
    public Object getEndArea() {
        return endArea;
    }

    @JsonProperty("end-area")
    public void setEndArea(Object endArea) {
        this.endArea = endArea;
    }

    @JsonProperty("ipis")
    public List<Object> getIpis() {
        return ipis;
    }

    @JsonProperty("ipis")
    public void setIpis(List<Object> ipis) {
        this.ipis = ipis;
    }

    @JsonProperty("type-id")
    public String getTypeId() {
        return typeId;
    }

    @JsonProperty("type-id")
    public void setTypeId(String typeId) {
        this.typeId = typeId;
    }

    @JsonProperty("life-span")
    public ArtistGetLifeSpan getLifeSpan() {
        return lifeSpan;
    }

    @JsonProperty("life-span")
    public void setLifeSpan(ArtistGetLifeSpan lifeSpan) {
        this.lifeSpan = lifeSpan;
    }

    @JsonProperty("isnis")
    public List<String> getIsnis() {
        return isnis;
    }

    @JsonProperty("isnis")
    public void setIsnis(List<String> isnis) {
        this.isnis = isnis;
    }

    @JsonProperty("country")
    public String getCountry() {
        return country;
    }

    @JsonProperty("country")
    public void setCountry(String country) {
        this.country = country;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/musicbrainz/artist/get/ArtistGet.kt
package uk.co.ourfriendirony.medianotifier.clients.musicbrainz.artist.get

import com.fasterxml.jackson.annotation.*

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonPropertyOrder(
    "disambiguation",
    "id",
    "area",
    "gender",
    "begin-area",
    "sort-name",
    "gender-id",
    "name",
    "type",
    "release-groups",
    "end-area",
    "ipis",
    "type-id",
    "life-span",
    "isnis",
    "country"
)
class ArtistGet {
    @get:JsonProperty("disambiguation")
    @set:JsonProperty("disambiguation")
    @JsonProperty("disambiguation")
    var disambiguation: String? = null

    @get:JsonProperty("id")
    @set:JsonProperty("id")
    @JsonProperty("id")
    var id: String? = null

    @get:JsonProperty("area")
    @set:JsonProperty("area")
    @JsonProperty("area")
    var area: ArtistGetArea? = null

    @get:JsonProperty("gender")
    @set:JsonProperty("gender")
    @JsonProperty("gender")
    var gender: Any? = null

    @get:JsonProperty("begin-area")
    @set:JsonProperty("begin-area")
    @JsonProperty("begin-area")
    var beginArea: ArtistGetBeginArea? = null

    @get:JsonProperty("sort-name")
    @set:JsonProperty("sort-name")
    @JsonProperty("sort-name")
    var sortName: String? = null

    @get:JsonProperty("gender-id")
    @set:JsonProperty("gender-id")
    @JsonProperty("gender-id")
    var genderId: Any? = null

    @get:JsonProperty("name")
    @set:JsonProperty("name")
    @JsonProperty("name")
    var name: String? = null

    @get:JsonProperty("type")
    @set:JsonProperty("type")
    @JsonProperty("type")
    var type: String? = null

    @get:JsonProperty("release-groups")
    @set:JsonProperty("release-groups")
    @JsonProperty("release-groups")
    var releaseGroups: List<ArtistGetReleaseGroup>? = null

    @get:JsonProperty("end-area")
    @set:JsonProperty("end-area")
    @JsonProperty("end-area")
    var endArea: Any? = null

    @get:JsonProperty("ipis")
    @set:JsonProperty("ipis")
    @JsonProperty("ipis")
    var ipis: List<Any>? = null

    @get:JsonProperty("type-id")
    @set:JsonProperty("type-id")
    @JsonProperty("type-id")
    var typeId: String? = null

    @get:JsonProperty("life-span")
    @set:JsonProperty("life-span")
    @JsonProperty("life-span")
    var lifeSpan: ArtistGetLifeSpan? = null

    @get:JsonProperty("isnis")
    @set:JsonProperty("isnis")
    @JsonProperty("isnis")
    var isnis: List<String>? = null

    @get:JsonProperty("country")
    @set:JsonProperty("country")
    @JsonProperty("country")
    var country: String? = null
}